core/data/src/test/java/uk/gov/gchq/gaffer/data/element/EdgeTest.java

```diff
@@ -290,6 +290,25 @@ public void shouldReturnFalseForEqualsWhenDirectedIdentifiersFlipped() {
     }
 
     @Test
+    public void shouldCreateUndirectedEdgesUsingNaturalOrdering() {
+        // Given
+        final Edge edge1 = new Edge("group");
+        edge1.setSource("source vertex");
+        edge1.setDestination("dest vertex");
+        edge1.setDirected(false);
+
+        // Given
+        final Edge edge2 = new Edge("group");
+        edge2.setSource("dest vertex");
+        edge2.setDestination("source vertex");
+        edge2.setDirected(false);
+
+        // Then
+        assertEquals(edge1, edge2);
+        assertEquals(edge1.toString(), edge2.toString());
```

Comment:
Is it worth adding a test that `edge1.getSource` equals `edge2.getSource`? The test of the `toString()` method partly tests that but it might be better to test it explicitly.